include/fiction/algorithms/physical_design/wiring_reduction.hpp

```diff
@@ -1044,70 +1064,114 @@ class wiring_reduction_impl
         // measure run time
         mockturtle::stopwatch stop{pst.time_total};
 
+        // record initial layout statistics
         pst.num_wires_before = plyt.num_wires() - plyt.num_pis() - plyt.num_pos();
         pst.x_size_before    = plyt.x() + 1;
         pst.y_size_before    = plyt.y() + 1;
 
+        // create an obstruction layout based on the original layout
         auto layout = obstruction_layout<Lyt>(plyt);
 
+        // initialize the list of wires to delete
         layout_coordinate_path<wiring_reduction_layout_type<coordinate<Lyt>>> to_delete = {};
 
         bool found_wires = true;
 
+        // lambda to update the timeout status and calculate remaining time
+        auto update_timeout = [&]() -> void
```

Comment:
Instead of a default capture (`[&]`), specify each name to capture separately. That prevents accidental copy where reference was intended and vice versa.